### PR TITLE
Updates MGT React getting started

### PR DESCRIPTION
**Pull request by waldekmastykarz**
Related to https://github.com/microsoftgraph/microsoft-graph-toolkit/issues/711

Updated Microsoft Graph Toolkit React docs to remove irrelevant pieces and include a step-by-step getting started walkthrough.

@nmetulev I included the auth piece for now in the doc. If we find a way to centralize it to simplify maintenance, we can replace it with a reference.

### CODE REVIEW

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 1]
```diff
@@ -7,147 +7,159 @@ author: elisenyang
...
+
+## Connect React app to Microsoft 365
+
+Now that we have registered our application with Azure Active Directory, we can connect the React app to Microsoft 365. We'll start with allowing users to log in to the app using their Microsoft account.
+
+1. In the Azure Portal, navigate to your application registration
+1. Ensure that you are on the **Overview** page
+1. From the **Essentials** section, copy the value of the **Application (client) ID** property
+   ![Application ID copied from the app registration overview page in the Azure Portal](../images/mgt-azure-add-appreg-copyappid.png)
+1. In the code editor open the **src/App.tsx** file, and modify its contents to the following:
```
Comment (nmetulev) [inline concepts/toolkit/get-started/use-toolkit-with-react.md:67]: I recommend adding the Provider initialization code in the index.tsx file since it's a one time initialization. Adding it in the functional component will cause the code to run every time the component re-renders.

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 2]
```diff
@@ -7,147 +7,159 @@ author: elisenyang
...
+
+      ```tsx
+      import { MsalProvider, Providers } from '@microsoft/mgt';
+      import { Login } from '@microsoft/mgt-react';
+      import React from 'react';
+      import './App.css';
+      
+      function App() {
+        Providers.globalProvider = new MsalProvider({
+          clientId: 'd7cb93c9-9097-4e38-8f06-7c0088ac3318'
```
Comment (nmetulev) [inline concepts/toolkit/get-started/use-toolkit-with-react.md:77]: Consider changing the clientId value to something like `[REPLACE_WITH_CLIENTID]` to make it a bit more clear that it needs to be replaced

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 3]
```diff
@@ -7,147 +7,159 @@ author: elisenyang
...
+      
+      export default App;
+      ```
+
+    We start, with configuring the authentication provider that should be used by Microsoft Graph Toolkit. In this case we use MSAL, which is a good default for building standalone applications. If you use any of the extensibility points in Microsoft 365, like Teams or SharePoint, you would use other providers.
+
+    Next, in the provider, we specify the ID of our Azure AD application.
+
+1. Replace the value of the `clientId` property with the value of the `Application (client) ID` property you copied previously in the Azure Portal
+1. Go back to the browser where your React app is running. You should now see a **Sign In** button
```
Comment (nmetulev) [inline concepts/toolkit/get-started/use-toolkit-with-react.md:97]: Consider adding a note that the code above also adds the `Login` component from the toolkit, which will render a sign in button

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 2 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:77]: I see this being used and I must admit that it often confuses me because I don't know if the whole value is a placeholder  or if the value should be put between brackets. This is why I prefer to specify the value directly to make it clear what format it's expected in.

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 1 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:67]: As soon as I move Provider initialization code to index.tsx, I'm getting the following error:

> Unhandled Rejection (Error): invalid key: msal.d7cb93c9-9097-4e38-8f06-7c0088ac3318.acquireTokenAccount|OTM2MGQ0ZDgtMzg0Yi00NGYxLWFlNmQtNTcyZjM2NjQzMzU4.ZjczMjIzODAtZjIwMy00MmZmLTkzZTgtNjZlMjY2ZjZkMmU0|a071d43a-952d-49ba-befb-a05ffc650514, The cached token key is not a valid JSON and cannot be parsed

The only way I can get it to work is to:
1. Launch the app
1. Login
1. Error
1. Clear local storage
1. Refresh page

I've compared it to your starter app and I'm not seeing any differences but perhaps I'm missing something trivial

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 4]
```diff
@@ -2,152 +2,166 @@
...
+
+Now that we have registered our application with Azure Active Directory, we can connect the React app to Microsoft 365. We'll start with allowing users to log in to the app using their Microsoft account.
+
+1. In the Azure Portal, navigate to your application registration
+1. Ensure that you are on the **Overview** page
+1. From the **Essentials** section, copy the value of the **Application (client) ID** property
+   ![Application ID copied from the app registration overview page in the Azure Portal](../images/mgt-azure-add-appreg-copyappid.png)
+1. In the code editor open the **src/App.tsx** file, and modify its contents to the following:
+
+      ```tsx
```
Comment (elisenyang) [inline concepts/toolkit/get-started/use-toolkit-with-react.md:69]: Thoughts on breaking this up into smaller bits of code to make it easier to digest? For example: import components, initialize provider, and add components

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 5]
```diff
@@ -2,152 +2,166 @@
...
+
+## Load data from Microsoft 365
+
+Microsoft Graph Toolkit not only simplifies authentication to Microsoft 365 but also loading its data. In this example, we'll show the signed in person's calendar.
+
+![Person's calendar displayed using Microsoft Graph Toolkit](../images/mgt-finished-app.png)
+
+1. In the code editor, change the contents of the **src/App.tsx** file to:
+
+    ```tsx
```
Comment (elisenyang) [inline concepts/toolkit/get-started/use-toolkit-with-react.md:114]: Here as well. Consider breaking this up into smaller steps

**Review by elisenyang — CHANGES_REQUESTED** [review]
Thanks so much for your help improving this guide! I just left a few comments.

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 2 above)
Comment (nmetulev) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:77]: Good point, what about just adding the text without the brackets:

`clientId: 'REPLACE_WITH_CLIENTID'`

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 1 above)
Comment (nmetulev) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:67]: Looks like this is a recent bug in the latest msal 1.x update: https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/2492. I think you are safe to ignore (assuming this will get fixed by the msal team)

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 6]
```diff
@@ -7,147 +7,159 @@ author: elisenyang
...
-  }
-
-  componentDidMount() {
-    this.refs.loginComponent.addEventListener('loginCompleted', e => {
-      // handle event
-    });
-  }
-}
-```
+To follow the steps below, you'll need a development machine with [Node.js](https://nodejs.org) and an account with Microsoft Azure. You can [get one for free](https://azure.microsoft.com/free/https://azure.microsoft.com/free/?WT.mc_id=m365-10340-wmastyka) if you don't have one yet.
```
[the remark this answers is not in the document]
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:20]: Ah, seems like I made a typo pasting it. Good catch! Sorry

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 4 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:69]: The trouble is that breaking it into smaller chunks makes it really hard to follow which code snippet should go where in the app. If we're to break up, we'd need to repeat the whole `App.tsx` each time. Since the code snippet isn't that large, I decided to include it just once. Ideally, we could clarify it by referring to line numbers but I don't think there is a way to show them, is there?

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 1 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:67]: _safe to ignore_ as in we keep Provider initialization in `index.tsx` and folks might get this error when running the sample too, or we keep provider initialization in `App.tsx` for now so that folks can build a working app and we'll update the sample and doc when the issue is fixed?

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 7]
```diff
@@ -2,152 +2,166 @@
...
+            </header>
+          </div>
+        );
+      }
+      
+      export default App;
+      ```
+
+    We start, with configuring the authentication provider that should be used by Microsoft Graph Toolkit. In this case we use MSAL, which is a good default for building standalone applications. If you use any of the extensibility points in Microsoft 365, like Teams or SharePoint, you would use other providers.
+
```
Comment (beth-panx) [inline concepts/toolkit/get-started/use-toolkit-with-react.md:93]: Should we link to the providers section from here?

**Review by beth-panx — APPROVED** [review]
Minor things I noticed. Looks great! Thank you for updating the docs 👍🏼

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 1 above)
Comment (nmetulev) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:67]: Looks like this was fixed in the latest [msal update](https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/2510) - do you mind trying it out again by making sure the latest msal get's pulled from npm?

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 8]
```diff
@@ -2,152 +2,166 @@
...
-```
-All properties and events map exactly as they are defined on the web component.
+> [!IMPORTANT]
+> Your app need to run on HTTPS in order to be able to authenticate against Microsoft 365. For local testing, with the React scripts, you can configure the local web server to run on HTTPS, by setting the `HTTPS` environment variable to `true`. You can do this either each time, by prefixing the `npm start` command with `HTTPS=true` or by setting the environment variable globally on your machine.
+
+You should be able to open your app in the browser via `https://localhost:3000`.
+
+![Newly created React app running in the browser](../images/mgt-base-react-app.png)
+
+[!INCLUDE [AAD with implicit flow app registration](../includes/aad-app-registration-spa.md)]
```
[the remark this answers is not in the document]
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:57]: This is a special syntax that will embed the steps on AAD as a part of the article rather than it being just a link that requires the reader to head off to a different place, complete the steps there and get back to the article.

File: concepts/toolkit/get-started/add-aad-app-registration.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+1. To create the app in Azure Active Directory, navigate to the Azure portal at https://portal.azure.com.
+1. From the menu, select **Azure Active Directory**
+   ![The 'Azure Active Directory' menu option highlighted in the Azure Portal](../images/aad-app-registration-spa/mgt-azure-menu-add.png)
+1. From the Azure Active Directory menu, select **App registrations**
+   ![The 'App registration' menu option highlighted in the Azure AD menu](../images/aad-app-registration-spa/mgt-azure-add-menu-appreg.png)
+1. From the top menu, select the **New registration** button
+   ![The 'New registration' button highlighted on the App registrations page in the Azure Portal](../images/aad-app-registration-spa/mgt-azure-add-appreg-newreg-button.png)
+1. Enter the name for your app, eg. `My M365 app`
+1. For the type of supported account types, select `Accounts in any organizational directory (Any Azure AD directory - Multitenant) and personal Microsoft accounts (e.g. Skype, Xbox)`
```
[the remark this answers is not in the document]
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/add-aad-app-registration.md:24]: Good idea. Let me see if I can find a resource that explains it

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 10]
```diff
@@ -2,152 +2,166 @@
...
+
+## Connect React app to Microsoft 365
+
+Now that we have registered our application with Azure Active Directory, we can connect the React app to Microsoft 365. We'll start with allowing users to log in to the app using their Microsoft account.
+
+1. In the Azure Portal, navigate to your application registration
+1. Ensure that you are on the **Overview** page
+1. From the **Essentials** section, copy the value of the **Application (client) ID** property
+   ![Application ID copied from the app registration overview page in the Azure Portal](../images/mgt-azure-add-appreg-copyappid.png)
```
[the remark this answers is not in the document]
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:66]: I'm using an MOD tenant that will self-destruct in a bit so no issue 😎

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 7 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:93]: Good idea! We can add a link to `other providers`.

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (beth-panx) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:57]: Gotcha!

File: concepts/toolkit/get-started/add-aad-app-registration.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+1. In the **Redirect URI** field, in the dropdown select `Web` and in the URL field enter `https://localhost:3000`
```
Comment (elisenyang) [inline concepts/toolkit/get-started/add-aad-app-registration.md:25]: I think this should be more general since not everyone will use something different. Something like "enter the URL at which you will be hosting your app (for example: https://localhost:3000/"

**Review by elisenyang — APPROVED** [review]

File: concepts/toolkit/get-started/add-aad-app-registration.md (hunk 11 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/add-aad-app-registration.md:25]: This URL comes from the React tools, so if you follow these steps to the letter, this will be your URL.

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 1 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:67]: It's working. I noticed though, that after moving Provider initialization to `index.tsx`, listening to Provider state change in `App.tsx` was no longer working. Instead, I had to use the `useIsSignedIn` function from your repo https://github.com/nmetulev/graph-react-starter/blob/87d6828d847c6ec98c72fdf70627a145ae5ed6f3/src/mgt.ts#L6

File: concepts/toolkit/get-started/add-aad-app-registration.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/add-aad-app-registration.md:24]: Link added

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 7 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:93]: Link added

File: concepts/toolkit/get-started/add-aad-app-registration.md (hunk 11 above)
Comment (elisenyang) [reply concepts/toolkit/get-started/add-aad-app-registration.md:25]: I think it would be great if we can reference this app registration page in other guides too, not just the React one! Developers may have different URL in those cases

**Review by nmetulev — APPROVED** [review]

File: concepts/toolkit/get-started/add-aad-app-registration.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+
+## Add new application registration in Azure Active Directory
+
+To create an application in Azure Active Directory, you need to add a new application registration, and then configure an app name and URL location.
+
+To create the app in Azure Active Directory:
+
+1. Go to the Azure portal at https://portal.azure.com.
+1. From the menu, select **Azure Active Directory**.
+   ![The 'Azure Active Directory' menu option highlighted in the Azure Portal](../images/aad-app-registration-spa/mgt-azure-menu-add.png)
```
Comment (Lauragra) [inline concepts/toolkit/get-started/add-aad-app-registration.md:20]: All the images in this article are gigantic and they also contain a lot of non-localizable text: https://review.docs.microsoft.com/en-us/graph/toolkit/get-started/add-aad-app-registration?branch=pr-en-us-10424. Unless the UI is difficult to follow for some reason, I recommend that you simply remove these screenshots.

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 13]
```diff
@@ -2,152 +2,319 @@
...
+> Your app need to run on HTTPS in order to be able to authenticate against Microsoft 365. For local testing, with the React scripts, you can configure the local web server to run on HTTPS by setting the `HTTPS` environment variable to `true`. You can do this either each time, by prefixing the `npm start` command with `HTTPS=true` (works only in bash) or by setting the environment variable globally on your computer.
 
-To use the components, first import the Toolkit in your application; `mgt-react` does not import the components automatically.
+You should be able to open your app in the browser via `https://localhost:3000`.
 
-```jsx
-import '@microsoft/mgt'
-```
-Now, you can import and use any of the components as regular React components. For example, to use the [Person component](../components/person.md), import it at the top of your file:
+![Newly created React app running in the browser](../images/mgt-base-react-app.png)
```
Comment (Lauragra) [inline concepts/toolkit/get-started/use-toolkit-with-react.md:55]: Is this image really adding anything? It's also pretty big on the page.

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 14]
```diff
@@ -2,152 +2,319 @@
...
 
-This topic covers how to use Microsoft Graph Toolkit with React. For a step-by-step walkthrough that describes how to create a new React application and use Microsoft Graph Toolkit, see [Using Microsoft Graph Toolkit with React](https://developer.microsoft.com/graph/blogs/a-lap-around-microsoft-graph-toolkit-day-13-using-microsoft-graph-toolkit-with-react/).
+If you're building apps with React, you can use the **@microsoft/mgt-react** package, which wraps MGT web components in React components and makes it easier to pass complex data.
 
-## Web Components and React
+This article describes the step-by-step process of using the Microsoft Graph Toolkit to create a React app and connect it to Microsoft 365. After completing the steps, you'll have a React app that shows the upcoming appointments of the currently signed in user from Microsoft 365.
 
-React passes all data to Custom Elements in the form of HTML attributes. For primitive data, this is fine, but it does not work when passing rich data, like objects or arrays. In those cases, you'll need to use a ref to pass in the object.
+![Finished app](../images/mgt-finished-app.png)
```
Comment (Lauragra) [inline concepts/toolkit/get-started/use-toolkit-with-react.md:16]: Suggest removing from here since we show the same image at the end of the article.

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 15]
```diff
@@ -2,152 +2,319 @@
...
-  };
+### Copy the Azure AD application registration ID
 
-  return <Person personDetails={personDetails}></Person>;
-};
-```
+1. In the Azure Portal, go to your application registration.
+1. Verify that you are on the **Overview** page.
+1. From the **Essentials** section, copy the value of the **Application (client) ID** property
+   ![Application ID copied from the app registration overview page in the Azure Portal](../images/mgt-azure-add-appreg-copyappid.png)
```
Comment (Lauragra) [inline concepts/toolkit/get-started/use-toolkit-with-react.md:68]: You can crop this image to only show the important part. Images with lots of English text can't be localized, and the content also isn't accessible to a screen reader. Try to avoid showing too much text in images.

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 16]
```diff
@@ -2,152 +2,319 @@
...
 
-#### Templates
+### Test signing in to your application
 
-All Microsoft Graph Toolkit components support [custom templates](../templates.md), which allow you to modify the content of a component. `mgt-react` allows you to use React to write templates for Microsoft Graph Toolkit components.
+You should now be able to sign in to your application with your Microsoft account.
 
-First, define a React component that uses the `MgtTemplateProps` object as its type of props, then use it as a child of the wrapped component.
+1. Go back to the browser where your React app is running. You should now see a **Sign In** button.
+   ![React app showing the Sign In button](../images/mgt-react-signin-button.png)
```
Comment (Lauragra) [inline concepts/toolkit/get-started/use-toolkit-with-react.md:166]: This image includes a lot of blank white space; please crop.

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 17]
```diff
@@ -2,152 +2,319 @@
...
+1. When you click the **Sign In** button, you will be prompted to sign in with your Microsoft account (you can use the same account as the one you accessed the Azure Portal with).
+1. Because this is the first time you're using this Azure AD application, you need to consent its use in your organization.
+1. After signing in, you will be redirected to your React app. Notice that the **Sign In** button changed to show your user's name
+  ![React app showing user info retrieved from Microsoft 365 using Microsoft Graph Toolkit](../images/mgt-react-userinfo.png).
```
Comment (Lauragra) [inline concepts/toolkit/get-started/use-toolkit-with-react.md:170]: This image can also be reduced.

File: concepts/toolkit/get-started/use-toolkit-with-react.md [hunk 18]
```diff
@@ -2,152 +2,319 @@
...
-```jsx
-import { Agenda } from 'mgt-react';
-
-const App = (props) => {
-  return <Agenda>
-    <MyEvent template="event">
-  </Agenda>
+Microsoft Graph Toolkit not only simplifies authentication to Microsoft 365, but also loading its data. In this example, you'll show the signed in person's calendar.
+
+![Person's calendar displayed using Microsoft Graph Toolkit](../images/mgt-finished-app.png)
```
Comment (Lauragra) [inline concepts/toolkit/get-started/use-toolkit-with-react.md:176]: This is another occurrence of the same image. Let's show it only once in the article; probably at the end is fine.

**Review by Lauragra — CHANGES_REQUESTED** [review]
Please see comments regarding the use of the images in the articles.

File: concepts/toolkit/get-started/add-aad-app-registration.md (hunk 12 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/add-aad-app-registration.md:20]: My idea was to help people unfamiliar with Azure orient themselves in the UI. You make a very good point though with regards to localization, I'll remove them.

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 13 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:55]: My idea was to help people ensure that we're starting from the same point. I'll remove it.

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 15 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:68]: My idea was to include just enough of the image to help people unfamiliar with Azure orient themselves in the UI. You make a good point about the amount of English text in the UI. I'm afraid though, that if we crop the image too much, its point will be lost and it will be even harder for people to find where in the UI they should be. Let's remove the image altogether.

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 16 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:166]: I'll remove the image altogether, as cropping it too much will remove some of the context (URL, browser window) with which the image will lose its value.

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 17 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:170]: Cropped

File: concepts/toolkit/get-started/use-toolkit-with-react.md (hunk 18 above)
Comment (waldekmastykarz) [reply concepts/toolkit/get-started/use-toolkit-with-react.md:176]: Removed

**Comment by waldekmastykarz** [comment]
Thank you for your feedback @Lauragra. I've just submitted a set of changes following your feedback.
